borg/key.py

```diff
@@ -126,28 +126,29 @@ class AESKeyBase(KeyBase):
     def id_hash(self, data):
         """Return HMAC hash using the "id" HMAC key
         """
-        return HMAC(self.id_key, data, sha256).digest()
+        return hmac_sha256(self.id_key, data)
 
     def encrypt(self, data):
         data = self.compressor.compress(data)
         self.enc_cipher.reset()
         data = b''.join((self.enc_cipher.iv[8:], self.enc_cipher.encrypt(data)))
```

Comment:
I'd say about one added line of Cython and one line shortened here? Performance difference a few percent at best.

I think the more important question is whether we want to mix AESKeyBase and crypot.AES on this level, as the code is now crypto.AES doesn't know and doesn't care where the IV comes from or how it's stored, this change would intertwine this implementation detail of AESKeyBase with crypto.AES, which is why I only commented instead of committing. (And what I probably should've remarked in the comment as well)